functionary/ui/views/tasks.py

```diff
@@ -188,3 +199,30 @@ def get(self, request, *args, **kwargs):
             )
 
         return render(request, "partials/task_result.html", context=context)
+
+
+@require_GET
+@login_required
+def get_task_log(request: HttpRequest) -> HttpResponse:
+    env = Environment.objects.get(id=request.session.get("environment_id"))
+    if not request.user.has_perm(Permission.TASK_READ, env):
+        return HttpResponseForbidden()
+
+    if not (task_id := request.GET.get("task_id")):
```

Comment:
The need for this check should go away if you change the path as suggesting in the comment on urls.py.